configure.ac

```diff
@@ -1696,6 +1700,37 @@ AC_SUBST(LAPACKINC)
 
 # Done checking LAPACK
 
+# libsvm option
+LIBSVM_INC=
+LIBSVM_LIB=
+USE_LIBSVM=
+
+LOC_CHECK_USE(libsvm,libsvm,USE_LIBSVM)
+
+if test -n "$USE_LIBSVM"; then
+  if test "${LIBSVM_INC}" ; then
+    CFLAGS_backup="${CFLAGS}"
+    CFLAGS="${LIBSVM_INC} ${CFLAGS}"
+    AC_CHECK_HEADER([svm.h], [], [
+      AC_MSG_ERROR([svm.h file not found in specified include path.])
+    ])
+    CFLAGS="${CFLAGS_backup}"
+  else
+    LOC_CHECK_INC_PATH(libsvm,LIBSVM,LIBSVM_INC)
+    LOC_CHECK_INCLUDES(svm.h,LIBSVM,$LIBSVM_INC, [
+      LOC_CHECK_INCLUDES(libsvm/svm.h,LIBSVM,$LIBSVM_INC)
+    ])
+  fi
+  LOC_CHECK_LIB_PATH(libsvm,libsvm,LIBSVM_LIB)
+  LOC_CHECK_LIBS(svm,svm_load_model,libsvm,$LIBSVM_LIB,LIBSVM_LIB,,,)
+  AC_DEFINE(HAVE_LIBSVM)
```

Comment:
> My suggested code is simpler and should be quite enough.

I commited your version as it seems to work just fine.

Reply:
The logic behind is quite simple:

´LOC_CHECK_INCLUDES` first check if it can `include <svm.h>` and defines `HAVE_LIBSVM`. If the first check fails it checks it can `include <libsvm/svm.h>` and defines `HAVE_LIBSVM_SVM_H`. That's all we need to know.